src/components/structures/MatrixChat.js

```diff
@@ -914,6 +915,14 @@ module.exports = React.createClass({
                 }
             }
         });
+
+        const krh = new KeyRequestHandler(cli);
+        cli.on("crypto.roomKeyRequest", (req) => {
```

Comment:
API: The `KeyRequestHandler` is being given the entire `MatrixClient`, and given we never call `handleKeyRequest` or `handleKeyRequestCancellation` in any other situation, might it be better to make it the responsibility of `KeyRequestHandler` to attach event listeners? No strong feelings either way, just a thought.